### ISSUE TITLE
Change 3bf169ee3b3c7476295cf3c9de0fe8fb760b503b introduced bug

### ISSUE THREAD

**Issue by satya-das** [wxFormBuilder/wxFormBuilder#518]
Changes done to replace use of Connect and Disconnect introduced an issue.
Now the bind is done with menu object instead of `this`  and so if I create a tool-button with same ID as that of menu item I need to add another Bind() (or Connect) call which is a regression because earlier I did not need to do that. Without adding another Bind() the tool-button click doesn't invoke event handler.

**Comment by satya-das** [comment]
I created a pull request to effectively revert the changes: https://github.com/wxFormBuilder/wxFormBuilder/pull/519

**Comment by sodevel** [comment]
AFAIK this is not a bug but was more like a design decision. The reason behind this was that the present way makes it easier to use the menu in different locations or move it around because the event handlers move with the menu. Binding at the parent element makes the menu only useable inside that parent.

**Comment by satya-das** [comment]
I don't think so. If menu needs to be moved around then it doesn't have to be created as child of frame (or any other window), just an independent menubar will work. Am I missing something? I have attached the image of independent menubar in wxFormBuilder. Please let me know if you still think it is not a bug.
![image](https://user-images.githubusercontent.com/6621459/58834947-e2165500-8654-11e9-8720-b9a6aaea22a8.png)

**Comment by satya-das** [comment]
A frame with menu and toolbar having same set commands is very common. This change breaks that common pattern for a case which has better alternative solution.

**Comment by jhasse** [comment]
> If menu needs to be moved around then it doesn't have to be created as child of frame (or any other window), just an independent menubar will work.

Can I insert this menubar into a frame in wxFormBuilder?

> Now the bind is done with menu object instead of this and so if I create a tool-button with same ID as that of menu item I need to add another Bind() (or Connect) call which is a regression because earlier I did not need to do that.

Can we insert this Bind automatically?

**Comment by satya-das** [comment]
I menubar can be inserted into a frame, not sure if it can be done using wxFB.

> Can we insert this Bind automatically?
I don't know how it can be done.

The problem with Bind approach is that a frame with both menu and toolbar having entry point for common commands will have to manually add Connect for all the commands. And this is a bigger issue.
